src/Relude/Extra/Foldable1.hs

```diff
@@ -139,6 +155,22 @@ instance Foldable1 NonEmpty where
     {-# INLINE maximum1 #-}
     {-# INLINE minimum1 #-}
 
+    maximumOn1 :: Ord b => (a -> b) -> NonEmpty a -> a
+    maximumOn1 func = foldl1' (cmpOn func)
+      where
+        cmpOn p a b = case p a `compare` p b of
```

Comment:
Could you, please add type signature to `cmpOn`? It usually improves code readability to add type signatures to functions in `where` :slightly_smiling_face: